Feat/medical agents

dimitri4d opened the pull request:
Initial draft of the medical agents using gradio, pydantic graph and shared state

aidev9 committed:
Initial commit: Multi-Agent Product Decision Graph with PydanticAI and DeepSeek R1

aidev9 committed:
Watch video https://youtu.be/r1pymNaji1E

dimitri4d committed:
initial commit

dimitri4d committed:
WIP models and agents

dimitri4d committed:
Pydantic-graph and Gradio UI implementation

dimitri4d committed:
Pydantic graph state workflow and parallel specialists

aidev9 commented:
@dimitri4d - couple of observations:
- Overall looks and works fantastic! I love the architecture and the organization of the project
- Poetry is an excellent choice for package management
- Agent graph tab is very helpful
- Question: Why put .env in the parent folder? Isn't it better to place it in the main folder? If so, let's update load_dotenv() in models.py. If not, that's OK, too. Let us update the README and give clear instructions where to put the .env file (folder)
- Let's rename the main folder to community-projects/pydantic-ai-medical-agents
- How do we add more agents - specialists? Can we add a section into README on how to
	- Add specialists
	- Customize specialists prompts

- Currently InitialNode calls all specialists in parallel:

```
 # Initialize specialist results
        specialties = ['cardiology', 'neurology', 'gastroenterology']
        ctx.state.specialist_results = {
            specialty: SpecialistResult(specialty=specialty)
            for specialty in specialties
        }
```
		
-When the system grows with more specialists, it may be more effective to narrow down the specialists based on the patient case. Can we add the option (and control it with a switch in the UI) for InitialNode to choose, based on the case presented, which specialists to call? For instance, a patient with chest pain will likely not need the opinion of a dermatologist
- Proposing to rename InitialNode to CoordinatorNode
- If CoordinatorNode is able to pick which agents to call, it would be nice to show the user which specialists are being consulted. Can we include an agent graph (under this image or additional tab) from a PydanticAI mermaid diagram, including the specialist agents? Or just a list of agents being called?
- How does the PDF or image processing work? Is this piece wired yet?
- Summary agent - is this wired and working yet? It displays the summary on the UI, but does it also generate a PDF report?

aidev9 reviewed (commented):
@dimitri4d - I provided some comments earlier. Please take a look and let me know if they make sense.

dimitri4d committed:
Refactor folder structure

dimitri4d committed:
Document upload and parsing, case preview, simplify patient model

dimitri4d committed:
Add new specialized medical agents

aidev9 commented:
@dimitri4d - does this need more reviewing or you're still adding features?

dimitri4d committed:
Add AgentSelector class for selecting relevant medical specialties

dimitri4d committed:
cleanup and move to /community-projects

dimitri4d commented:
@aidev9 - Ready for review. I added the document upload, more specialists, agent coordinator, improved readme and moved to community projects folder. The only functionality not included is a PDF generations of the summary
